pkg/controllers/updaterun/controller.go

```diff
@@ -284,78 +284,155 @@ func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
+				"Invalid object type", "object", klog.KObj(oldObj))
+			return
+		}
+		newClusterAppReq, ok := newObj.(*placementv1beta1.ClusterApprovalRequest)
+		if !ok {
+			klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("cannot cast runtime object to ClusterApprovalRequest")),
+				"Invalid object type", "object", klog.KObj(newObj))
+			return
+		}
+		oldAppReq = oldClusterAppReq
+		newAppReq = newClusterAppReq
+	} else {
+		oldNamespacedAppReq, ok := oldObj.(*placementv1beta1.ApprovalRequest)
+		if !ok {
+			klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("cannot cast runtime object to ApprovalRequest")),
+				"Invalid object type", "object", klog.KObj(oldObj))
+			return
+		}
+		newNamespacedAppReq, ok := newObj.(*placementv1beta1.ApprovalRequest)
+		if !ok {
+			klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("cannot cast runtime object to ApprovalRequest")),
+				"Invalid object type", "object", klog.KObj(newObj))
+			return
+		}
+		oldAppReq = oldNamespacedAppReq
+		newAppReq = newNamespacedAppReq
 	}
 
-	approvedInOld := condition.IsConditionStatusTrue(meta.FindStatusCondition(oldAppReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApproved)), oldAppReq.Generation)
-	approvedInNew := condition.IsConditionStatusTrue(meta.FindStatusCondition(newAppReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApproved)), newAppReq.Generation)
+	approvedInOld := condition.IsConditionStatusTrue(meta.FindStatusCondition(oldAppReq.GetApprovalRequestStatus().Conditions, string(placementv1beta1.ApprovalRequestConditionApproved)), oldAppReq.GetGeneration())
+	approvedInNew := condition.IsConditionStatusTrue(meta.FindStatusCondition(newAppReq.GetApprovalRequestStatus().Conditions, string(placementv1beta1.ApprovalRequestConditionApproved)), newAppReq.GetGeneration())
 
 	if approvedInOld == approvedInNew {
-		klog.V(2).InfoS("The approval status is not changed, ignore queueing", "clusterApprovalRequest", klog.KObj(newAppReq))
+		klog.V(2).InfoS("The approval status is not changed, ignore queueing", "approvalRequestObj", klog.KObj(newAppReq))
 		return
 	}
 
-	updateRun := newAppReq.Spec.TargetUpdateRun
+	updateRun := newAppReq.GetApprovalRequestSpec().TargetUpdateRun
 	if len(updateRun) == 0 {
-		klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("TargetUpdateRun field in ClusterApprovalRequest is empty")),
-			"Invalid clusterApprovalRequest", "clusterApprovalRequest", klog.KObj(newAppReq))
+		klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("TargetUpdateRun field in ApprovalRequest is empty")),
+			"Invalid approval request", "approvalRequestObj", klog.KObj(newAppReq))
 		return
 	}
+
 	// enqueue to the updaterun controller queue.
-	q.Add(reconcile.Request{
-		NamespacedName: types.NamespacedName{Name: updateRun},
-	})
+	if isClusterScoped {
```

Comment:
No need the this if statement. `newAppReq.GetNamespace()` would return "" for clusterApprovalRequests.